sklearn/tree/_utils.pxd

```diff
@@ -39,6 +40,7 @@ ctypedef fused realloc_ptr:
     (DOUBLE_t*)
     (DOUBLE_t**)
     (Node*)
+    (Cell*)
```

Comment:
This registers the `Cell` arrays in the fused type used for reshaping the arrays.  
This could be factored out in `sklearn.utils` but for now we simply used it as it is.